src/libraries/System.Net.Mail/src/System/Net/Mail/MailAddressCollection.cs

```diff
@@ -39,7 +39,7 @@ protected override void InsertItem(int index, MailAddress item)
 
         internal void ParseValue(string addresses)
         {
-            IList<MailAddress> result = MailAddressParser.ParseMultipleAddresses(addresses);
+            List<MailAddress> result = MailAddressParser.ParseMultipleAddresses(addresses);
```

Comment:
I'm surprised we're not currently getting a diagnostic about this.

Reply:
Is there even such a diagnostic?

Reply:
Yes:
https://github.com/dotnet/runtime/blob/c08faf9216976a14f06a11373fbd3aec7671bf7a/eng/CodeAnalysis.src.globalconfig#L468

Looks like we're not configuring it yet to pick up internal use, just its default of private.

Reply:
I am tempted to dig deeper, but I am not sure where to start. Any ideas? Otherwise I will merge this PR and leave it alone.

Reply:
I'm not sure. @buyaa-n, do you know why CA1859 wouldn't flag this?

Reply:
I don't know, looks worth filing an issue in roslyn-analyzers repo. CC @geeknoid

Reply:
I filed https://github.com/dotnet/roslyn-analyzers/issues/7078